python_scripts/linear_models.py

```diff
@@ -565,9 +572,13 @@ def f(x):
 # the regressor, scikit-learn provides a `Pipeline`, which 'chains' the
 # transformer and regressor together. The pipeline allows you to use a
 # sequence of transformer(s) followed by a regressor or a classifier, in one
-# call. This pipeline exposes the same API as the regressor and classifier
+# call. (i.e. fitting the pipeline will fit both the transformer(s) and the regressor. 
+# Then predicting from the pipeline will first transform the data through the transformer(s)
+# then predict with the regressor from the transformed data)
+
+# This pipeline exposes the same API as the regressor and classifier
 # and will manage the calls to `fit` and `transform` for you, avoiding any
-# problems with data leakage.
+# problems with data leakage (like fitting the transformer on the test data).
```

Comment:
I think we should introduce concept by itself and not tagged on. I think this isn't enough information

Reply:
Maybe a reference to its entry in the glossary then ?

Reply:
I copied the entry of glossary into the text ^^
https://github.com/lucyleeow/scikit-learn-mooc/pull/2/commits/bdc49fcd3ea3ac7670b6cb468c88826cdf53c9cb

Reply:
Is there a glossary? And my suggestion wasn't for this, just that it would be nice to explain the concept somewhere in the notebooks, not necessarily here. If we do explain it, there is no need to go too in depth here. Anyway this is fine